components/server/src/server.rs

```diff
@@ -279,6 +281,23 @@ where
         );
         let pd_client =
             Self::connect_to_pd_cluster(&mut config, env.clone(), Arc::clone(&security_mgr));
+        // check if TiKV need to run in snapshot recovery mode
+        let is_recovering_marked = pd_client
+            .is_recovering_marked()
+            .expect("failed to get recovery mode from PD");
```

Comment:
This is causing compatibility issue between tikv and v6.2.0 pd.

Reply:
checked with tikv QA, it looks tikv 6.3 can not connect to pd v6.2.0, so we will fix this in master branch. not a block issue for 6.3.0.

Reply:
fixed https://github.com/tikv/tikv/issues/13497